### ISSUE TITLE
[clang] Support unnamed parameters in -Wdocumentation

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#172470]
Currently `-Wdocumentation` does not support unnamed parameters, since it requires that a name is present. For example:

```cpp
/// @param some unnamed parameter
/// @param y something 
void f(int, int y) {}
```

It fails with:

```cpp
<source>:1:12: warning: parameter 'some' not found in the function declaration [-Wdocumentation]
    1 | /// @param some unnamed parameter
      |            ^~~~
```
https://godbolt.org/z/efqxdTTjb

It would be great if this use case could be supported in some way. In a relevant [Doxygen thread](https://github.com/doxygen/doxygen/issues/6926) they propose using an integer to identify the unnamed parameter:

```cpp
/// @param 1 some unnamed parameter
```

**Comment by evelez7** [comment]
So I noticed that someone mentioned that `\param .` doesn't trigger a doxygen warning for a non-existent parameter. That seems like a solution, ~~but the issue is still open. We don't want to deviate from doxygen but I want to make sure this is intended behavior.~~

The respective Doxygen issue has since been updated by the developer showing support for syntax to document positional and unnamed parameters.

This has since been merged as a feature, so clang shouldn't warn on this.
